### PR TITLE
feat: operator hold by partition facet

### PR DESCRIPTION
**Pull request by jaime-iobermudez**
This pull request refactors the handling of operator-initiated holds by splitting the `operatorCreateHoldByPartition` functionality out of the monolithic `HoldManagementFacet` into its own dedicated facet, `OperatorHoldByPartitionFacet`. This results in a cleaner separation of concerns, simplifies the interface hierarchy, and updates all relevant configuration and test files to use the new facet. The change is non-breaking, as the selector for `operatorCreateHoldByPartition` remains unchanged, ensuring backward compatibility.

**Core contract and interface refactor:**

- Extracted `operatorCreateHoldByPartition` from the now-removed `HoldManagementFacet` into a new `OperatorHoldByPartitionFacet`, with its own interface (`IOperatorHoldByPartition`) and abstract contract (`OperatorHoldByPartition`). The resolver key `_OPERATOR_HOLD_BY_PARTITION_RESOLVER_KEY` was added for registration. (`[[1]](diffhunk://#diff-184cda64df0a6bb391592d75bb9d91468ecb15c12a7469e96a16b41f4081b99dR1-R30)`, `[[2]](diffhunk://#diff-56fccc19cd8ec267575f7687737f6de46e6848ead28618b5fc860f6634a2de89R1-R51)`, `[[3]](diffhunk://#diff-4f80b8a0c520079b60a3eba732362eda7a853c42dcba9f30c5a6723246e0f53cR1-R41)`, `[[4]](diffhunk://#diff-5d5e7d57e7df0d88b9d56a817db4dcfd22aba2c2bd27030409a9e501bde02567R1-R37)`, `[[5]](diffhunk://#diff-a6761447814cc412fd1b009f087d5b68cc0a5e3ad8ef1dd4e34fdd2e6b3b465dR240-R242)`)
- Removed `IHoldManagement`, `HoldManagement`, and `HoldManagementFacet` contracts and cleaned up all imports and interface aggregations that referenced them. (`[[1]](diffhunk://#diff-f99662d36d7ae96285f62ce780315b33c225e3dae3a905e918fc52765d4cc504L1-L77)`, `[[2]](diffhunk://#diff-ffa3fd0a9c462b77f05ee42cc12690e9e9ca7d40c5800b9787adb6f18d194684L1-L26)`, `[[3]](diffhunk://#diff-3000948afadc698734956a87dc404ba8c641a2420bafd00290caa30a109b8275L1-L20)`, `[[4]](diffhunk://#diff-4ecd2b22860d79dbe9636ccb2aeefa5c123aeabec9f8332a7e7a816e9839fd92L5-R10)`, `[[5]](diffhunk://#diff-40dbee73667a1c027b07f3aba73c9a5f76a07c2e6cb6c8b88fe4201d84435d75L139-R140)`)

**Interface and inheritance updates:**

- Updated `IAsset` to inherit from `IOperatorHoldByPartition` instead of indirectly via `IHoldManagement`, and removed the aggregation of `IHoldManagement` from `IHold`. (`[[1]](diffhunk://#diff-40dbee73667a1c027b07f3aba73c9a5f76a07c2e6cb6c8b88fe4201d84435d75R47)`, `[[2]](diffhunk://#diff-40dbee73667a1c027b07f3aba73c9a5f76a07c2e6cb6c8b88fe4201d84435d75R163)`, `[[3]](diffhunk://#diff-4ecd2b22860d79dbe9636ccb2aeefa5c123aeabec9f8332a7e7a816e9839fd92L5-R10)`)

**Configuration and script changes:**

- Replaced all instances of `HoldManagementFacet` with `OperatorHoldByPartitionFacet` in configuration files and domain-specific `createConfiguration.ts` scripts (equity, bond, bondFixedRate, bondKpiLinkedRate, bondSustainabilityPerformanceTargetRate, loan, loanPortfolio). (`[[1]](diffhunk://#diff-409dd60ab11b542fadbe019ebe0bae2d62b9a88197c5df9c72dbf6139790278bL130-R130)`, `[[2]](diffhunk://#diff-a720a90e2027c51c86c52a8b0a2c5f1f55b10a397357c77f3b78e8e5414bd2c5L115-R115)`, `[[3]](diffhunk://#diff-340f40bde9e011844c32605590792d04e8a08c309f2dd013580aaaf0422862beL110-R110)`, `[[4]](diffhunk://#diff-be815552a2fcc781130fb3b3076ffadbe135f6b2346fba9a91c36461ce03703eL105-R105)`, `[[5]](diffhunk://#diff-07d24066b881af524d31b71601198e8e15abd4b8d151b0ffd700c81292d687aeL105-R105)`, `[[6]](diffhunk://#diff-d31ee2f421823c23b554f15e2b40af208cca35fa7b27905fa5d54d6c48d1d06eL106-R106)`, `[[7]](diffhunk://#diff-2c85f3ee7a2c03e0b4a796d6e735fe1cfd5f5aaa1e4e4ccf0f9c1f0b3250df08L102-R102)`, `[[8]](diffhunk://#diff-e333092a268307cc17506ff1435c7caf6512ac6a460d0ad8bbfea1b87fd481baL109-R109)`)

**Test and library updates:**

- Added comprehensive tests for `OperatorHoldByPartitionFacet` covering all modifiers, happy paths, and edge cases. Removed time travel test facet for the old `HoldManagementFacet`. (`[[1]](diffhunk://#diff-184cda64df0a6bb391592d75bb9d91468ecb15c12a7469e96a16b41f4081b99dR1-R30)`, `[[2]](diffhunk://#diff-1b79f8f7b9fa6dfa282d6ec12e705be05502d3e2f2373ae06cee44482f8f5ec0L1-L9)`)
- Updated orchestrator libraries to remove mappings for `HoldManagementFacet` and add `OperatorHoldByPartitionFacet`. (`[.changeset/maf-operator-hold-by-partition-facet.mdR1-R30](diffhunk://#diff-184cda64df0a6bb391592d75bb9d91468ecb15c12a7469e96a16b41f4081b99dR1-R30)`)

**Backward compatibility:**

- The 4-byte selector for `operatorCreateHoldByPartition` remains unchanged, so any calls through `IAsset` or other interfaces continue to work as before. (`[.changeset/maf-operator-hold-by-partition-facet.mdR1-R30](diffhunk://#diff-184cda64df0a6bb391592d75bb9d91468ecb15c12a7469e96a16b41f4081b99dR1-R30)`)

## Type of change

- [ ] Bug fix 🐞
- [ ] New feature ✨
- [ ] Breaking change 💥
- [ ] Documentation update 📖
- [ ] Refactor 🔧

## Testing

**Node version**:

- [ ] 20
- [ ] 22
- [ ] 24

## Checklist

- [ ] Style Guidelines followed ✅
- [ ] Documentation Updated 📚
- [ ] **Linters** - No New Warnings ⚠️
- [ ] Local Tests Pass ✅
- [ ] Effective Tests Added ✔️
- [ ] No reduction of **Coverage** ✅

### CODE REVIEW

File: packages/ats/contracts/contracts/facets/layer_1/hold/IHold.sol [hunk 1]
```diff
@@ -2,10 +2,9 @@
...
-import { IHoldManagement } from "./IHoldManagement.sol";
 
-/// @title IHold — off-chain umbrella for the Hold Management domain
-/// @notice Aggregates IHoldManagement into a TypeChain-friendly handle for tests and SDK consumers.
+/// @title IHold — off-chain umbrella for the Hold domain
+/// @notice Aggregates hold types into a TypeChain-friendly handle for tests and SDK consumers.
 /// @dev DO NOT inherit this interface from any facet contract. Each facet must `is` exactly its
 ///      per-facet interface. This umbrella contains no function declarations of its own.
-interface IHold is IHoldTypes, IHoldManagement {}
+interface IHold is IHoldTypes {}
```
Comment (marcosio) [inline packages/ats/contracts/contracts/facets/layer_1/hold/IHold.sol]: Why empty interface? Where it is used?

**Review by marcosio — CHANGES_REQUESTED** [review]

**Review by marcosio — APPROVED** [review]
